### PR TITLE
xmoji: init at 0.5.1

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #333443

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/xm/xmoji/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    libXcursor
+    xcbutilimage
+  ]);
+
+  makeFlagsArray = [ "prefix=${placeholder "out"}" ];
+
+  meta = {
+    description = "Plain X11 emoji keyboard";
+    homepage = "https://github.com/Zirias/xmoji";
+    lib = lib.licenses.bsd2;
```
Comment (AndersonTorres) [inline pkgs/by-name/xm/xmoji/package.nix:57]: _You meant to say_

```suggestion
    license = lib.licenses.bsd2;
```

? --  Google (TM) Search

File: pkgs/by-name/xm/xmoji/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    fontconfig
+    harfbuzz
+    libpng
+    libxkbcommon
+    xcb-util-cursor
+  ] ++ (with xorg; [
+    xcbutil
+    libXcursor
+    xcbutilimage
+  ]);
```
Comment (eclairevoyant) [inline pkgs/by-name/xm/xmoji/package.nix:50]:
```suggestion
    xcbutil
    libXcursor
    xcbutilimage
  ]
```
attrs under `xorg` can be passed in directly

File: pkgs/by-name/xm/xmoji/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pkg-config,
+  fontconfig,
+  harfbuzz,
+  libpng,
+  xorg,
```
Comment (eclairevoyant) [inline pkgs/by-name/xm/xmoji/package.nix:9]:
```suggestion
  xcbutil,
  libXcursor,
  xcbutilimage,
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/xm/xmoji/package.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/by-name/xm/xmoji/package.nix]: Is there any recommendation for abolishing `xorg` this way?

Comment (eclairevoyant) [reply pkgs/by-name/xm/xmoji/package.nix]: Makes it easier to override, as you don't have to override the whole package set.
And CI complained about formatting, which was obviated by not passing xorg at all.

**Review by drupol — APPROVED** [review]

**Pull request merged**